### DISCUSSION TITLE
Multiple USB Cameras - matching data from udevadm walkthroughs

### DISCUSSION THREAD

**Discussion by TangoOversway** [paukstelis/octoprint_deploy#47] [General]
I am currently using a PiCam and a USB cam. I have found this USB cam has exactly the same ID info as another USB camera I was using. I did udevadm `info -q all -n <devnode> --attribute-walk|sort > output_x.txt` and then used diff to compare both files. The output for both USB cams was identical. For several reasons, I'd like to replace my PiCam with a 2nd USB cam that's the same model as the USB cam I'm using now on my 2nd printer. However, using the setup I'm using (haven't converted to Octoprint_Deploy yet), I find I can't make any udev rules that can distinguish my old and new USB cams from each other.

Does this project have any way to tell two such USB cameras apart? Even the serial numbers match. I did see you can specify to ignore the serial numbers, but if all the info is the same, I don't see how that's going ot help.

**Comment by paukstelis** [comment]
Yes, USB cameras can be detected by port or serial number. If the serial
numbers are the same you can run the script with the noserial option which
will force cameras to use the USB port assignment.

**Reply by TangoOversway** [reply]
Don't port assignments sometimes change on reboot?

I thought using udev rules to determine a camera by info the camera returns was, in part, because the port assignment could change on reboot.

**Comment by paukstelis** [comment]
Physical ports do not change on reboot.

**Reply by TangoOversway** [reply]
> Physical ports do not change on reboot.

Thanks. So as long as I keep them plugged in to the same physical connector, there's no problem?

I never knew that it was possible to find the physical port on the software side. So now I understand and learned something new. Thank you!
